src/zepben/evolve/services/network/tracing/networktrace/network_trace_step_path_provider.py

```diff
@@ -0,0 +1,273 @@
+#  Copyright 2025 Zeppelin Bend Pty Ltd
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from __future__ import annotations
+
+import itertools
+import sys
+from typing import Generator, Optional, Callable, Iterable, List, Union, Type, TYPE_CHECKING
+
+from zepben.evolve.model.cim.iec61970.base.wires.clamp import Clamp
+from zepben.evolve.model.cim.iec61970.base.wires.connectors import BusbarSection
+from zepben.evolve.model.cim.iec61970.base.wires.cut import Cut
+from zepben.evolve.model.cim.iec61970.base.core.terminal import Terminal
+from zepben.evolve.model.cim.iec61970.base.wires.aclinesegment import AcLineSegment
+from zepben.evolve.services.network.tracing.connectivity.terminal_connectivity_connected import TerminalConnectivityConnected
+from zepben.evolve.services.network.tracing.networktrace.network_trace_step import NetworkTraceStep
+
+if TYPE_CHECKING:
+    from zepben.evolve.services.network.tracing.networktrace.operators.network_state_operators import NetworkStateOperators
+
+__all__ = ['NetworkTraceStepPathProvider']
+
+PathFactory = Callable[[Terminal, AcLineSegment], Optional[NetworkTraceStep.Path]]
+
+
+class NetworkTraceStepPathProvider:
+    def __init__(self, state_operators: Type[NetworkStateOperators]):
+        self.state_operators = state_operators
+
+    def next_paths(self, path: NetworkTraceStep.Path) -> Generator[NetworkTraceStep.Path, None, None]:
+        path_factory = self._create_path_with_phases_factory(path) if path.nominal_phase_paths else self._create_path_factory(path)
+
+        def _get_next_paths():
+            to_equipment = path.to_equipment
+            if isinstance(to_equipment, AcLineSegment):
+                return self._next_paths_from_ac_line_segment(to_equipment, path, path_factory)
+            elif isinstance(to_equipment, BusbarSection):
+                return self._next_paths_from_busbar(path, path_factory)
+            elif isinstance(to_equipment, Clamp):
+                return self._next_paths_from_clamp(to_equipment, path, path_factory)
+            elif isinstance(to_equipment, Cut):
+                return self._next_paths_from_cut(to_equipment, path, path_factory)
+            else:
+                if path.traced_internally:
+                    return self._next_external_paths(path, path_factory)
+                else:
+                    return seq_term_map_to_path(path.to_terminal.other_terminals(), path_factory)
```

Comment:
Suggestion: No need to have the nested if that was required for the Kotlin when syntax.
```suggestion
            elif path.traced_internally:
                return self._next_external_paths(path, path_factory)
            else:
                return seq_term_map_to_path(path.to_terminal.other_terminals(), path_factory)
```